jerry-core/parser/js/js-lexer.c

```diff
@@ -1429,11 +1429,17 @@ lexer_parse_number (parser_context_t *context_p) /**< context */
       }
 
       lexer_check_numbers (context_p, &source_p, source_end_p, LIT_CHAR_7, true);
-    }
-    else if (source_p[1] >= LIT_CHAR_8
-             && source_p[1] <= LIT_CHAR_9)
-    {
-      parser_raise_error (context_p, PARSER_ERR_INVALID_NUMBER);
+
+      if (source_p <= source_end_p && source_p[0] >= LIT_CHAR_8
```

Comment:
This syntax is invalid. Please put every check into different line. Also `source_p < source_end_p`